diff --git a/ext/cpu_features/src/hwcaps.c b/src/hwcaps.c
index f44f6c3..e857d35 100644
--- a/ext/cpu_features/src/hwcaps.c
+++ b/ext/cpu_features/src/hwcaps.c
@@ -60,7 +60,8 @@ const char* CpuFeatures_GetBasePlatformPointer(void);
 #define AT_PLATFORM 15
 #define AT_BASE_PLATFORM 24
 
-#if defined(HAVE_STRONG_GETAUXVAL)
+#if defined(HAVE_STRONG_GETAUXVAL) && !defined(HAVE_LIBNX) && \
+    !defined(__SWITCH__)
 #include <sys/auxv.h>
 static unsigned long GetElfHwcapFromGetauxval(uint32_t hwcap_type) {
   return getauxval(hwcap_type);
@@ -101,7 +102,7 @@ static uint32_t GetElfHwcapFromGetauxval(uint32_t hwcap_type) {
   return ret;
 }
 #else
-#error "This platform does not provide hardware capabilities."
+static unsigned long GetElfHwcapFromGetauxval(uint32_t hwcap_type) { return 0; }
 #endif
 
 // Implementation of GetHardwareCapabilities for OS that provide
